Experimenting with i386 compilation

arcticoder opened the pull request:
- added VSC workspace to gitignore
- some improvements to cpp.properties
- Dockerfile improvements: added clang
- Note clang options don't get picked up by compiler-explorer for some reason
- TODO: fix compiler error, missing library when including <cstdint>

Commit:
Experimenting with i386 compilation

- added VSC workspace to gitignore
- some improvements to cpp.properties
- Dockerfile improvements: added clang
- Note clang options don't get picked up by compiler-explorer for some reason
- TODO: fix compiler error, missing library when including <cstdint>

arcticoder commented:
Referencing https://www.youtube.com/watch?v=zBkNBP00wJE

madduci commented on Dockerfile:
```diff
@@ -15,6 +15,8 @@ RUN echo "*** Installing Compiler Explorer ***" \
         nodejs \
         make \
         git \
+        clang-11 \
```
@arcticoder can you please remove the following? Actually the compilers are being installed in the parent image. I will try to add it, as also clang 12

madduci commented on cpp.properties:
```diff
@@ -20,8 +20,10 @@ compiler.clang10.exe=/usr/bin/clang++-10
+compiler.clang11.options=-std=c++1z -Wall -Wextra -Wshadow -O3 -m32 -march=i386
 compiler.clangdefault.exe=/usr/bin/clang++-11
 compiler.clangdefault.name=clang default (11)
+compiler.clangdefault.options=-std=c++1z -Wall -Wextra -Wshadow -O3 -m32 -march=i386
```
by doing so, aren't we obliging everyone to use 32 bit output? Could we make it optional? Or something like:

```
compiler.clang11x86.exe=/usr/bin/clang++-11
compiler.clang11x86.name=clang 11 x86
compiler.clang11x86.options=-std=c++1z -Wall -Wextra -Wshadow -O3 -m32 -march=i386
```

madduci commented:
Hi @arcticoder 

thanks for submitting a pull request. I've just added a couple of comments. The idea is welcome, but as for the changes, I believe they are then forcing everyone who pull this image to use Clang 11 on 32 bit. A better approach would be give a separated option for it.

Commit:
cpp.properties, Dockerfile

- removed clang-11 as it's being installed in parent image as advised by @madduci
- making i386 compilation optional (clang11x86) as advised by @madduci

madduci commented on Dockerfile:
```diff
@@ -15,6 +15,8 @@ RUN echo "*** Installing Compiler Explorer ***" \
         make \
         git \
+        clang-11 \
+        libstdc++6 \
```
we don't need this one either?

arcticoder replied:
I can remove it, just FYI I was getting errors when it wasn't included. The container was missing the libraries.

madduci replied:
I'll add it to the parent image then!

Commit:
Dockerfile

- Removed libstdc++6 as requested by @madduci
